put the actual description first in issue templates

mezzarobba opened the pull request:
...so that relevant information appears when hovering over items of the
issue list, and so that one can read the actual issue on the issue page
without scrolling over a full page of boilerplate.

Also remove the title from the Description section of the PR template
(same rationale).

Any opinions on that?

<!-- ^^^^^
Please provide a concise, informative and self-explanatory title.
Don't put issue numbers in there, do this in the PR body below.
For example, instead of "Fixes #1234" use "Introduce new method to calculate 1+1"
-->
### 📚 Description

<!-- Describe your changes here in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Closes #1337" -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ ] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### ⌛ Dependencies
<!-- List all open pull requests that this PR logically depends on -->
<!--
- #xyz: short description why this is a dependency
- #abc: ...
-->

mezzarobba committed:
put the actual description first in issue templates

...so that relevant information appears when hovering over items of the
issue list, and so that one can read the actual issue on the issue page
without scrolling over a full page of boilerplate.

Also remove the title from the Description section of the PR template
(same rationale).

mezzarobba committed:
encourage PR submitters to remove checkboxes...

...that don't make sense with their particular PR

dimpase reviewed (approved):
I agree that the checkboxes should not come first, so it looks better now.

tobiasdiez commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -14,6 +12,7 @@ For example, instead of "Fixes #1234" use "Introduce new method to calculate 1+1
 <!-- Put an `x` in all the boxes that apply. -->
 <!-- If your change requires a documentation PR, please link it appropriately -->
 <!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+<!-- Feel free to remove irrelevant items. -->
```
Not sure about this one. The idea was that you simply don't tick the item if it does not apply. This then might help reviewers as red flags, e.g. "I should double check that writing tests indeed doesn't make sense for this PR".

mezzarobba replied:
I agree that this is debatable, but if you just leave them unchecked there is not difference between “not done yet” and “not applicable”. (And it can interfere with the use of actual tasks list in the rest of the issue, as github will show the corresponding items as remaining tasks.) In any case, this is just a detail, and I'm fine with dropping this commit if you insist.

tobiasdiez replied:
I don't have any strong feelings about this either.

tobiasdiez reviewed (commented):
From a dev point of view that mainly reads issues, this seems like a good and helpful change. The rationale for the checklist-first approach is that it serves as a guideline for people creating issues. If the checklist is at the end, there is a high chance that it is simply ignored (are you going to really search for similar issues if you already spent 5 mins writing the main issue body?)

Not sure what is more important...

mezzarobba committed:
issue templates: move the instructions back to the beginning

...while keeping the checklist at the end.

With this version, I thought it made sense to apply the change to
ftbfs issues too. I left the "Environment" section near the beginning in
that case, though it makes for a small inconsistency wrt the bug report
template.

tobiasdiez reviewed (approved):
Thanks, LGTM

dimpase committed:
Merge branch 'develop' into issue_template

mezzarobba committed:
Merge branch 'develop' into issue_template
